integration_test/secret_storage_test.dart

```diff
@@ -0,0 +1,136 @@
+      }
+      if (Hive.isBoxOpen(WebSecretStorageProvider.encryptedBoxTable)) {
+        await Hive.lazyBox(WebSecretStorageProvider.encryptedBoxTable).clear();
+      }
+    } else {
+      const mobileStorage = FlutterSecureStorage(aOptions: AndroidOptions(encryptedSharedPreferences: true));
+      await mobileStorage.deleteAll();
+    }
+  });
+
+  tearDownAll(() async {
+    if (kIsWeb) {
+      try {
+        // [Hive.deleteFromDisk] doesn't close by itself on
+        // web https://github.com/isar/hive/pull/734.
+        await Hive.deleteFromDisk().timeout(const Duration(seconds: 2));
+      } catch (_) {}
+    }
+  });
+
+  // Runs before other test since box should
+  if (kIsWeb) {
+    testWidgets("Given no password, when read, then throw exception", (widgetTester) async {
+      // Arrange
+      const String key = "foo";
+
+      // Act
+      try {
+        await storage.read(key);
+        fail("Read should have triggered exception");
+      } on SecretStorageException catch (e) {
+        // Assert
+        expect(e.error, SecretStorageError.encryptedBoxNotOpened);
+      } catch (e) {
+        fail("Exception should not be of type ${e.runtimeType}");
+      }
+    });
+  }
+
+  testWidgets("When set, read and delete from storage, then update storage correctly", (widgetTester) async {
+    // Arrange
+    await storage.setPassword("password");
+    const String key = "foo";
+    const String value = "bar";
+
+    // Act
+    await storage.set(key, value);
+    final beforeDelete = await storage.read(key);
+    await storage.delete(key);
+    final afterDelete = await storage.read(key);
+
+    // Assert
+    expect(beforeDelete, isNotNull);
+    expect(afterDelete, isNull);
+    expect(beforeDelete, value);
+  });
+
+  testWidgets("Given no password set, when check if exist, then return false", (widgetTester) async {
+    // Act
+    final actual = await storage.hasPassword();
+
+    // Assert
+    expect(actual, false);
+  });
+
+  testWidgets("When set password, then password set and one can unlock", (widgetTester) async {
+    // Arrange
+    const String password = "foobar";
+
+    // Act
+    final hasPasswordBefore = await storage.hasPassword();
+    await storage.setPassword(password);
+    final hasPasswordAfter = await storage.hasPassword();
+    final succeeded = await storage.unlock(password);
+
+    // Assert
+    expect(hasPasswordBefore, false);
+    expect(hasPasswordAfter, true);
+    expect(succeeded, true);
+  });
+
+  testWidgets("When unlock with wrong password, then return false", (widgetTester) async {
+    // Arrange
+    const String password = "foobar";
+    await storage.setPassword(password);
+
+    // Act
+    final unlocked = await storage.unlock("somethingElse");
+
+    // Assert
+    expect(unlocked, false);
+  });
+
+  testWidgets("Given no password set, when trying to unlock, then throw exception", (widgetTester) async {
+    try {
```

Comment:
I think it would make it a bit more readable if we do something along the lines of this (then I don't need to concern myself with understanding the different paths that lead to a failed test)

```suggestion
    dynamic expectedError;
    try {
      // Acts
      await storage.unlock("password");
    } on SecretStorageException catch (e) {
      // Assert
      expectedError = e;
    }
    expect(expectedError, isA<SecretStorageException>());
    expect((expectedError as SecretStorageException).error, SecretStorageError.noPassword);
```